README.md

```diff
@@ -62,17 +72,48 @@ Username: postgres
+
+8) Navigate to the frontend directory: `./autoscheduler/frontend/src/` and run `npm install`
+    - This is for TypeScript compilation and testing.
+    
+9) Before making code changes, please go to the github project [wiki](https://github.com/aggie-coding-club/Rev-Registration/wiki) and follow the Project Setup and Install section.
 
 ## Running
 
 Before running any commands, if you're not running in the virtual environment(you should see `(env)` somewhere in your current terminal line), run `source env/bin/activate` or `./env/Scripts/activate`.
 
-You will have to make and apply migrations before running the server (and whenever our models are changed). To generate the files that django uses to apply migrations, run `./manage.py makemigrations`, and to then apply these to the database, run `./manage.py migrate`.
+**Server:**
+- You will have to make and apply migrations before running the server (and whenever our models are changed). To generate the files that django uses to apply migrations, run `./manage.py makemigrations`, and to then apply these to the database, run `./manage.py migrate`.
+    - `manage.py` can be found in the `./autoscheduler/` directory.
+
+- After migrating, you can run the server with `./manage.py runserver`. When running for the first time, you will have to configure `autoscheduler/autoscheduler/config/postgres-info.json` with your postgres username and password. This file will be automatically created if you run `./manage.py runserver`.
+
+- If `./manage.py` isn't working, try to use `python manage.py ...` or `python3 manage.py ...` or `py manage.py ...` in place of it.
+    - See python aliases above
+
+- The first time you run the server, you will need to scrape the course info from TAMU
+    - `python manage.py scrape_depts`
+    - `python manage.py scrape_courses`
+    - `python manage.py scrape_grades`
+        - You can put a term afterwards if you just want to scrape a specific term for quick testing:
+            - `python manage.py scrape_courses 202011`
+
+**Frontend:**
+- Navigate to the frontend source directory: `cd ./autoscheduler/frontend/src` and run `npm run dev`
+    - This sets up a watch for file changes and compiles them
+    - ctrl-c to quit
+
+**`localhost:8000`**
+- Open in web browser after doing the above steps.
 
-After migrating, you can run the server with `./manage.py runserver`. When running for the first time, you will have to configure `autoscheduler/autoscheduler/config/postgres-info.json` with your postgres username and password. This file will be automatically created if you run `./manage.py runserver`.
+**Testing:**
 
-If `./manage.py` isn't working, try to use `python manage.py` or `python3 manage.py` in place of it.
+- The frontend has automated test suites which can be run with:
+    - `cd ./autoscheduler/frontend/src`
+    - `npm run test`
+        - Can put filename after to test only one file: `npm run test testfile.test.tsx`
+            - No path
```

Comment:
```suggestion
        - Can put filename after to test only one file: `npm run test testfile.test.tsx`
```
You actually can do the path (e.g. `npm run test tests/ui/SchedulingPage.test.tsx`) so this comment isn't necessary, but it is great to know cause I had no idea you could do that

Reply:
does that work? I got a bunch of errors about it not being able to find the test file when I did that

Reply:
I added a little text blurb about it in the last commit.